okhttp/src/commonJvmAndroid/kotlin/okhttp3/CompressionInterceptor.kt

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2025 Block, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package okhttp3
+
+import okhttp3.ResponseBody.Companion.asResponseBody
+import okhttp3.internal.http.promisesBody
+import okio.BufferedSource
+import okio.GzipSource
+import okio.Source
+import okio.buffer
+
+/**
+ * Transparent Compressed response support.
+ *
+ * Adds Accept-Encoding to request and checks (and strips) Content-Encoding in
+ * responses. n.b. this replaces the transparent gzip compression in BridgeInterceptor so should generally include
+ * gzip in the list of algorithms.
+ */
+open class CompressionInterceptor(
```

Comment:
@swankjesse the name is a bit weird, but I wonder whether this should support configurable request compression also?